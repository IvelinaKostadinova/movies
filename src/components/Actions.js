import React from 'react';

const Actions = () => {
  return (
    <div className="actions-container">
      <button className="filter-btn" type="submit">
        ALL
      </button>
      <button className="filter-btn" type="submit">
        DOCUMENTARY
      </button>
      <button className="filter-btn" type="submit">
        COMEDY
      </button>
      <button className="filter-btn" type="submit">
        HORROR
      </button>
      <button className="filter-btn" type="submit">
        CRIME
      </button>
      <button className="filter-btn" type="submit">
        FANTASY
      </button>

      <select name="sort" id="sort-slc">
        <option value="releseDate">RELEASE DATE</option>
        <option value="title">TITLE</option>
      </select>
      <label id="sort-lbl">SORT BY</label>
    </div>
  );
};

export default Actions;
