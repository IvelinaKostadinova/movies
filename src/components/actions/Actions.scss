@import '../../styles/variables';

.actions {
    width: 100%;
    height: 6rem;
    background-color: $black;

    &__sort {
      float: right;
      margin: 3.2rem;
      background-color: transparent;
      color: $white;
      border-color: transparent;
      font-family: $font-family;
      font-size: 1.8rem;
      cursor: pointer;
    }

    &__sort__label {
      float:right;
      margin: 3rem;
      background-color: transparent;
      color: $white;
      border-color: transparent;
      font-family: $font-family;
      font-size: 1.8rem;
    }
}
