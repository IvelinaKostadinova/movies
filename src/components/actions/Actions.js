import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Filter from '../filter/Filter';

import './Actions.scss';

const Actions = ({ initialSelectedGenre, initialSort, onFilter, onSort }) => {
  const [selectedGenre, setSelectedGenre] = useState(initialSelectedGenre);

  const handleSelect = (e) => {
    onSort(e.target.value);
  };

  return (
    <div className="actions">
      {['ALL', 'DOCUMENTARY', 'COMEDY', 'HORROR', 'CRIME', 'FANTASY'].map(
        (genre) => {
          return (
            <Filter
              key={genre}
              text={genre}
              handleClick={(e) => {
                setSelectedGenre(genre);
                if (e == 'ALL') e = '';
                onFilter(e);
              }}
              active={genre === selectedGenre}
            ></Filter>
          );
        }
      )}

      <select
        defaultValue={initialSort}
        name="sort"
        className="actions__sort"
        onChange={handleSelect}
      >
        <option value="release_date">RELEASE DATE</option>
        <option value="title">TITLE</option>
      </select>
      <label className="actions__sort__label">SORT BY</label>
    </div>
  );
};

Actions.propTypes = {
  initialSelectedGenre: PropTypes.string.isRequired,
  initialSort: PropTypes.string.isRequired,
  onFilter: PropTypes.func.isRequired,
  onSort: PropTypes.func.isRequired,
};

export default Actions;
