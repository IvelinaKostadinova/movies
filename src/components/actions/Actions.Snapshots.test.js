import React from 'react';
import Actions from './Actions';
import renderer from 'react-test-renderer';

it('sets all actions', () => {
  const tree = renderer.create(
    <Actions onFilter={jest.fn()} onSort={jest.fn()} />
  );

  expect(tree).toMatchSnapshot();
});
