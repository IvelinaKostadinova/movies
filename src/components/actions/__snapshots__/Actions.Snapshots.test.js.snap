// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets all actions 1`] = `
<div
  className="actions"
>
  <button
    className="filter"
    onClick={[Function]}
    type="submit"
  >
    ALL
  </button>
  <button
    className="filter"
    onClick={[Function]}
    type="submit"
  >
    DOCUMENTARY
  </button>
  <button
    className="filter"
    onClick={[Function]}
    type="submit"
  >
    COMEDY
  </button>
  <button
    className="filter"
    onClick={[Function]}
    type="submit"
  >
    HORROR
  </button>
  <button
    className="filter"
    onClick={[Function]}
    type="submit"
  >
    CRIME
  </button>
  <button
    className="filter"
    onClick={[Function]}
    type="submit"
  >
    FANTASY
  </button>
  <select
    className="actions__sort"
    name="sort"
    onChange={[Function]}
  >
    <option
      value="release_date"
    >
      RELEASE DATE
    </option>
    <option
      value="title"
    >
      TITLE
    </option>
  </select>
  <label
    className="actions__sort__label"
  >
    SORT BY
  </label>
</div>
`;
